packages/flutter_tools/lib/src/plugins.dart

```diff
@@ -155,8 +158,18 @@ void _writeAndroidPluginRegistrant(String directory, List<Plugin> plugins) {
     'plugins': androidPlugins,
   };
 
-  final String javaSourcePath = fs.path.join(directory, 'src', 'main', 'java');
-  final String registryPath = fs.path.join(javaSourcePath, 'io', 'flutter', 'plugins', 'GeneratedPluginRegistrant.java');
+  final String javaSourcePath = fs.path.join(
+    (await project.androidPluginRegistrantDirectory).path,
+    'src',
+    'main',
+    'java',
+  );
+  final String registryPath = fs.path.join(javaSourcePath,
```

Comment:
[nit] Move `javaSourcePath,` to the next line.

Reply:
Done